### PR TITLE
Debug bundle

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

Implements the ability to generate a debug bundle in Broker.

Debug bundles are saved in the CWD of the Broker process, and saved in the following scenarios:
- When running `broker fix` if errors are found.
- When running `broker fix --export-bundle`.

## Acceptance criteria

Debug bundles are collected when running `broker fix`, and contain:
- Traces
- FOSSA CLI debug bundles (these are uncompressed and prettified before being included in the bundle)

## Testing plan

Automated tests cover this, but I also manually tested and the results were as expected.

## References

https://fossa.atlassian.net/browse/ANE-840

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).

### CODE REVIEW

File: src/debug.rs [hunk 1]
```diff
@@ -51,6 +60,19 @@ pub enum ValidationError {
     RetentionBelowMinimum,
 }
 
+/// Export mode for the debug bundle.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum BundleExport {
+    /// Bundle export is disabled entirely.
+    Disable,
```
Comment (spatten) [inline src/debug.rs:67]: It looks like `BundleExport::Disable` is only used in tests right now. Is this something we want to allow users to turn on via config? I can't really think of a reason for this, but wanted to double check

File: src/ext/io/sync.rs [hunk 2]
```diff
@@ -65,6 +71,99 @@ pub enum Error {
...
+#[tracing::instrument]
+pub fn copy_debug_bundle<P, Q>(file: P, rel: Q) -> Result<(NamedTempFile, PathBuf), Report<Error>>
+where
+    P: AsRef<Path> + std::fmt::Debug,
+    Q: AsRef<Path> + std::fmt::Debug,
+{
+    const EXT: &str = ".json.gz";
+    if !file.as_ref().to_string_lossy().ends_with(&EXT) {
+        let copy = copy_temp(file)?;
+        return (copy, rel.as_ref().to_path_buf()).wrap_ok();
```
Comment (spatten) [inline src/ext/io/sync.rs:154]: In a previous review, you mentioned that early returns from if statements were considered unidiomatic. I'm thinking that this case is a good counter-example to that, as the code in the `else` block would be too long. Is that the difference?

**Review by spatten — APPROVED** [review]
:+1:

File: src/debug.rs (hunk 1 above)
Comment (jssblck) [reply src/debug.rs:67]: I don't think there's any reason to allow users to set this.

File: src/ext/io/sync.rs (hunk 2 above)
Comment (jssblck) [reply src/ext/io/sync.rs:154]: Yep, I opted for early return here because the bulk of the function would be in an `else` block otherwise!

**Pull request merged**